pkg/skaffold/generate_pipeline/profile.go

```diff
@@ -32,55 +32,49 @@ import (
 	yamlv2 "gopkg.in/yaml.v2"
 )
 
-func CreateSkaffoldProfile(out io.Writer, config *latest.SkaffoldConfig, configFile string) error {
+func CreateSkaffoldProfile(out io.Writer, config *latest.SkaffoldConfig, configFile string) (*latest.Profile, error) {
 	reader := bufio.NewReader(os.Stdin)
 
+	// Check for existing oncluster profile, if none exists then prompt to create one
 	color.Default.Fprintln(out, "Checking for oncluster skaffold profile...")
-	profileExists := false
 	for _, profile := range config.Profiles {
 		if profile.Name == "oncluster" {
-			profileExists = true
-			break
+			color.Default.Fprintln(out, "profile \"oncluster\" found")
+			return &profile, nil
 		}
 	}
 
-	// Check for existing oncluster profile, if none exists then prompt to create one
-	if profileExists {
-		color.Default.Fprintln(out, "profile \"oncluster\" found!")
-		return nil
-	}
-
 confirmLoop:
 	for {
 		color.Default.Fprintf(out, "No profile \"oncluster\" found. Create one? [y/n]: ")
 		response, err := reader.ReadString('\n')
 		if err != nil {
```

Comment:
Not in this PR, but later you can add a func to re-try getting user input 3 times before failing.
It would be great if you create an issue in github and assign it your self with priority P2/